### PR TITLE
Implement the progress bar component

### PR DESCRIPTION
**Pull request by sandhose**
This also includes a story which shows how to use it for a password strength indicator.

Design reference: https://www.figma.com/design/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?node-id=5425-6829&t=vGtJ0aBTjajPgRzf-4

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -129,5 +130,6 @@
   "resolutions": {
     "jackspeak": "2.1.1",
     "@radix-ui/react-id": "1.1.0"
-  }
+  },
+  "packageManager": "yarn@1.22.19+sha1.4ba7fc5c6e704fce2066ecbfb0b0d8976fe62447"
```
Comment (t3chguy) [inline package.json:134]: This seems a little too precise, what if I have 

```
 ~  yarn --version                                                                                                                                 479ms  Wed 10 Jul 13:33:52 2024
1.22.22
```

Also seems unrelated to this PR

File: src/components/Progress/Progress.module.css [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    var(--cpd-progress-bar-main) 50%,
+    var(--cpd-progress-bar-muted) 50%,
+    var(--cpd-progress-bar-muted) 75%,
+    var(--cpd-progress-bar-main) 75%,
+    var(--cpd-progress-bar-main) 100%,
+    var(--cpd-progress-bar-muted) 100%
+  );
+
+  /* sqrt(number of stripes * 2 * (stripe width)^2) = sqrt(4 * 2 * 2^2) = sqrt(32) */
+  background-size: 5.6569px 5.6569px;
```
Comment (t3chguy) [inline src/components/Progress/Progress.module.css:86]: That is hellish

File: src/components/Progress/Progress.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+import { Root, Indicator } from "@radix-ui/react-progress";
+import useId from "../../utils/useId";
+
+type ProgressProps = {
+  /* The size variant of the progress bar */
+  size: "sm" | "lg";
+
+  /* The color variant to use for the progres bar indicator */
+  color?: "red" | "orange" | "lime" | "green";
```
Comment (t3chguy) [inline src/components/Progress/Progress.tsx:30]: What happens if you don't specify a colour?

Also we're a British company, colour*
image

and `progress*`

File: package.json (hunk 1 above)
Comment (sandhose) [reply package.json:134]: grr, that's my local yarn which added this itself for some reason :(
I think it's corepack related or something?

File: src/components/Progress/Progress.module.css (hunk 2 above)
Comment (sandhose) [reply src/components/Progress/Progress.module.css:86]: I'm very sorry about this :(

File: src/components/Progress/Progress.tsx (hunk 3 above)
Comment (sandhose) [reply src/components/Progress/Progress.tsx:30]: It becomes gray. You probably should only use this for empty state, as this also propagates to the label on top. See [the password strength story](https://quenting-progress-bar.compound-web.pages.dev/?path=/docs/form-password-form--docs#empty)

And damn I never noticed the britishness of props. But why are the token using `color` then? 🤔 

Also, https://github.com/hashanp/postcss-spiffing

Comment (t3chguy) [reply src/components/Progress/Progress.tsx:30]: > It becomes gray. You probably should only use this for empty state, as this also propagates to the label on top.

Can the default be commented plz

> But why are the token using color then?

No idea

> https://github.com/hashanp/postcss-spiffing

This is awesome lol

Comment (sandhose) [reply src/components/Progress/Progress.tsx]: I'm avoiding the problem by renaming the prop `tint` if you're alright with that 😬 

I also tweaked the comments, and made sure they get parsed as JSDoc comments, so that they render in the IDE and Storybook

File: package.json (hunk 1 above)
Comment (sandhose) [reply package.json]: Removed

**Review by t3chguy — APPROVED** [review]
